iceoryx_posh/include/iceoryx_posh/internal/popo/smart_chunk.hpp

```diff
@@ -42,7 +42,7 @@ struct SmartChunkPrivateData
     SmartChunkPrivateData& operator=(const SmartChunkPrivateData&) = delete;
     ~SmartChunkPrivateData() = default;
 
-    cxx::unique_ptr<T> smartChunkUniquePtr;
+    cxx::optional<cxx::unique_ptr<T>> smartChunkUniquePtr;
```

Comment:
Do we already have an issue to make the smart chunk also non-nullable?

Reply:
I created one: https://github.com/eclipse-iceoryx/iceoryx/issues/1654